lib/galaxy/managers/context.py

```diff
@@ -256,7 +256,7 @@ def user_can_do_run_as(self) -> bool:
 
     @property
     def user_is_active(self) -> bool:
-        return not self.app.config.user_activation_on or self.user is None or self.user.active
+        return not self.app.config.user_activation_on or (self.user is not None and self.user.active)
```

Comment:
I think it was on purpose that anonymous users are exempt from this check, as they are subject to different limitations.

Reply:
Oh I see, this is a different case then. Is it ok if we add a function here that throws an exception for logged out/anonymous user instead? (or maybe we already have one)

Reply:
That should be the `anonymous` property defined above. But again I would not check that inside `user_is_active` / `check_user_activation()`.